src/inverted_index.c

```diff
@@ -756,49 +756,61 @@ IndexDecoderProcs InvertedIndex_GetDecoder(uint32_t flags) {
   }
 }
 
-IndexReader *NewNumericReader(IndexSpec* sp, InvertedIndex *idx, const NumericFilter *flt) {
+IndexReader *NewNumericReader(IndexSpec *sp, InvertedIndex *idx, const NumericFilter *flt) {
   RSIndexResult *res = NewNumericResult();
   res->freq = 1;
   res->fieldMask = RS_FIELDMASK_ALL;
   res->num.value = 0;
 
   IndexDecoderCtx ctx = {.ptr = (void *)flt};
-  return NewIndexReaderGeneric(sp, idx, readNumeric, ctx, res, 1);
+  IndexDecoderProcs procs = {.decoder = readNumeric};
+  return NewIndexReaderGeneric(sp, idx, procs, ctx, res, 1);
 }
 
-typedef struct IndexCriteriaTesterCtx{
-  char* term;
+static t_docId calculateId(t_docId lastId, uint32_t delta, int isFirst) {
+  t_docId ret;
+
+  if (isFirst && delta != 0) {
+    // this is an old version rdb, the first entry is the docid itself and
+    // not the delta
+    ret = delta;
+  } else {
+    ret = delta + lastId;
+  }
+  return ret;
+}
+
+typedef struct IR_CriteriaTester {
```

Comment:
Don't need the struct name here, just the typedef.

Reply:
I actually like it this way, but I will remove it if you so against it...

Reply:
I'm not *super* against it.